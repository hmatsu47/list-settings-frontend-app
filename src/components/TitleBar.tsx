import AppBar from "@suid/material/AppBar";
import Box from "@suid/material/Box";
import IconButton from "@suid/material/IconButton";
import Stack from "@suid/material/Stack";
import Typography from "@suid/material/Typography";
import ReplayIcon from "@suid/icons-material/Replay";
import {
  fetchLocal,
  fetchRemote,
  fetchUriSettings,
} from "../api/fetchUriSettings";
import { fetchTagSettings } from "../api/fetchTagSettings";

const reload = () => {
  fetchUriSettings(fetchLocal);
  fetchUriSettings(fetchRemote);
  fetchTagSettings();
};

export const TitleBar = () => {
  return (
    <AppBar position="static">
      <Stack direction="row">
        <Typography
          variant="h6"
          component="div"
          sx={{
            flexGrow: 1,
            padding: "0 0 0 10px",
          }}
        >
          コンテナリリースイメージ一覧
        </Typography>
        <Box title="再読み込み">
          <IconButton
            size="small"
            edge="start"
            color="inherit"
            aria-label="reload"
            sx={{ mr: 1 }}
            onClick={() => reload()}
          >
            <ReplayIcon sx={{ fontSize: "large", padding: "2px" }} />
            Reload
          </IconButton>
        </Box>
      </Stack>
    </AppBar>
  );
};
