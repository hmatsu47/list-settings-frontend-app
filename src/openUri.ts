export const openUriInNewTab = (uri: string) => {
  window.open(uri, "_blank", "noopener,noreferrer");
};