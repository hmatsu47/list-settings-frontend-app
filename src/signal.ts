import { createSignal } from "solid-js";
import { AlertColor } from "@suid/material/Alert";

import { TagSetting, UriSetting } from "./type";

export const [services, setServices] =
  createSignal<string[] | null | undefined>();
export const [remoteServices, setRemoteServices] =
  createSignal<string[] | null | undefined>();
export const [tagSettings, setTagSettings] =
  createSignal<TagSetting[] | null | undefined>();
export const [uriSettings, setUriSettings] =
  createSignal<UriSetting[] | null | undefined>();
export const [uriRemoteSettings, setUriRemoteSettings] =
  createSignal<UriSetting[] | null | undefined>();
export const [message, setMessage] = createSignal<string | undefined>();
export const [messageSeverity, setMessageSeverity] =
  createSignal<AlertColor | undefined>();
