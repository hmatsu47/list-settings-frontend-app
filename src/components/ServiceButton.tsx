import Button from "@suid/material/Button";
import { service, setService } from "../signal";

type Props = {
  serviceName: string;
};

export const ServiceButton = (props: Props) => {
  return (
    <Button
      variant={service() === props.serviceName ? "contained" : "outlined"}
      size="small"
      color={service() === props.serviceName ? "primary" : "inherit"}
      onClick={(e) => {
        setService(props.serviceName);
        // 選択肢をローカルストレージに記録しておく
        localStorage.setItem("selectedService", props.serviceName);
      }}
      sx={{ textTransform: "none" }}
      title={`${props.serviceName} を選択`}
    >
      {props.serviceName}
    </Button>
  );
};
