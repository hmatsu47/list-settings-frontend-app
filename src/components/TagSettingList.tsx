import { For, Show } from "solid-js";
import Box from "@suid/material/Box";
import Button from "@suid/material/Button";
import Paper from "@suid/material/Paper";
import Stack from "@suid/material/Stack";
import Table from "@suid/material/Table";
import TableBody from "@suid/material/TableBody";
import TableCell from "@suid/material/TableCell";
import TableContainer from "@suid/material/TableContainer";
import TableHead from "@suid/material/TableHead";
import TableRow from "@suid/material/TableRow";
import Typography from "@suid/material/Typography";
import { openUriInNewTab } from "../openUri";
import { tagSettings } from "../signal";

const convertEnvironmentName = (environmentName: string) => {
  switch (environmentName) {
    case "dev":
      return "開発";
    case "prod":
      return "本番";
  }
  return environmentName;
};

const buttonColor = (environmentName: string) => {
  const colors = localStorage.getItem("tagButtonColor");
  if (!colors) {
    return "#616161";
  }
  const map = new Map<string, string>(Object.entries(JSON.parse(colors)));
  return map.get(environmentName) ?? "#616161";
};

export const TagSettingList = () => {
  const headerTitle = localStorage.getItem("tagSettingsHeaderTitle");
  const uriPrefix = localStorage.getItem("tagSettingUriPrefix");
  const uriSuffix = localStorage.getItem("tagSettingUriSuffix");

  return (
    <Show when={tagSettings() !== undefined} fallback={<></>}>
      <Box
        sx={{
          width: "100%",
          minWidth: "1024px",
          display: "flex",
        }}
        aria-live="polite"
      >
        <Stack spacing={2} direction="column">
          <Typography variant="h5">{headerTitle}</Typography>
          <Show
            when={tagSettings() !== null}
            fallback={
              <Typography variant="h6">
                リポジトリにイメージがありません
              </Typography>
            }
          >
            <TableContainer component={Paper}>
              <Table aria-label="simple table">
                <TableHead>
                  <TableRow>
                    <TableCell>設定</TableCell>
                    <TableCell>リリース対象イメージタグ</TableCell>
                  </TableRow>
                </TableHead>
                <TableBody>
                  <For each={tagSettings()} fallback={<></>}>
                    {(settingItem) => (
                      <TableRow>
                        <TableCell component="th" scope="row">
                          <Button
                            variant="contained"
                            size="small"
                            // color="primary"
                            onClick={() => {
                              openUriInNewTab(
                                `${uriPrefix}${settingItem.environment_name}${uriSuffix}`
                              );
                            }}
                            sx={{
                              textTransform: "none",
                              backgroundColor: `${buttonColor(
                                settingItem.environment_name
                              )}`,
                            }}
                            title={`${convertEnvironmentName(
                              settingItem.environment_name
                            )}環境のリリース設定画面を開く`}
                          >
                            {convertEnvironmentName(
                              settingItem.environment_name
                            )}
                            環境設定
                          </Button>
                        </TableCell>
                        <TableCell>{settingItem.tags.join(", ")}</TableCell>
                      </TableRow>
                    )}
                  </For>
                </TableBody>
              </Table>
            </TableContainer>
          </Show>
        </Stack>
      </Box>
    </Show>
  );
};
