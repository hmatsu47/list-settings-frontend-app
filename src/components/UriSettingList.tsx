import { For, Show } from "solid-js";
import Box from "@suid/material/Box";
import Button from "@suid/material/Button";
import Paper from "@suid/material/Paper";
import Table from "@suid/material/Table";
import TableBody from "@suid/material/TableBody";
import TableCell from "@suid/material/TableCell";
import TableContainer from "@suid/material/TableContainer";
import TableHead from "@suid/material/TableHead";
import TableRow from "@suid/material/TableRow";
import Typography from "@suid/material/Typography";
import { service, uriRemoteSettings, uriSettings } from "../signal";
import { formatDateTimeDisplay } from "../formatDateTime";
import { UriSetting } from "../type";
import { openUriInNewTab } from "../openUri";

const convertEnvironmentName = (environmentName: string) => {
  switch (environmentName) {
    case "stg":
      return "ステージング";
    case "prod":
      return "本番";
  }
  return environmentName;
};

export const UriSettingList = () => {
  const headerTitle = localStorage.getItem("uriSettingsHeaderTitle");

  return (
    <Show
      when={
        (service() && uriRemoteSettings() !== undefined) ||
        uriSettings() !== undefined
      }
      fallback={<></>}
    >
      <Box
        sx={{
          width: "100%",
          minWidth: "1024px",
          display: "flex",
        }}
        aria-live="polite"
      >
        <Typography variant="h3">{headerTitle}</Typography>
        <Show
          when={uriRemoteSettings() !== null && uriSettings() !== null}
          fallback={
            <Typography variant="h6">
              リポジトリにイメージがありません
            </Typography>
          }
        >
          <TableContainer component={Paper}>
            <Table aria-label="simple table">
              <TableHead>
                <TableRow>
                  <TableCell>設定</TableCell>
                  <TableCell>イメージURI</TableCell>
                  <TableCell>リリース日時</TableCell>
                </TableRow>
              </TableHead>
              <TableBody>
                <SettingListParts settings={uriRemoteSettings()} />
                <SettingListParts settings={uriSettings()} />
              </TableBody>
            </Table>
          </TableContainer>
        </Show>
      </Box>
    </Show>
  );
};

type Props = {
  settings: UriSetting[] | null | undefined;
};

const SettingListParts = (props: Props) => {
  const uriPrefix = localStorage.getItem("uriSettingUriPrefix");
  const uriSuffix = localStorage.getItem("uriSettingUriSuffix");

  return (
    <Show when={props.settings} fallback={<></>}>
      <Box
        sx={{
          width: "100%",
          minWidth: "1024px",
          display: "flex",
        }}
        aria-live="polite"
      >
        <For each={props.settings} fallback={<></>}>
          {(settingItem) => (
            <Show
              when={settingItem.service_name === service()}
              fallback={<></>}
            >
              <TableRow>
                <TableCell component="th" scope="row">
                  <Button
                    variant="contained"
                    size="small"
                    color="primary"
                    onClick={() => {
                      openUriInNewTab(
                        `${uriPrefix}${settingItem.environment_name}${uriSuffix}`
                      );
                    }}
                    sx={{ textTransform: "none" }}
                    title={`${convertEnvironmentName(
                      settingItem.environment_name
                    )}環境のリリース設定画面を開く`}
                  >
                    {convertEnvironmentName(settingItem.environment_name)}
                    環境設定
                  </Button>
                </TableCell>
                <TableCell>
                  {settingItem.next_release
                    ? settingItem.next_release.image_uri
                    : settingItem.last_released.image_uri
                    ? settingItem.last_released.image_uri
                    : ""}
                </TableCell>
                <TableCell>
                  {settingItem.next_release
                    ? `次回：${formatDateTimeDisplay(
                        new Date(settingItem.next_release.release_at)
                      )}`
                    : settingItem.last_released
                    ? `前回：${formatDateTimeDisplay(
                        new Date(settingItem.last_released.released_at)
                      )}`
                    : ""}
                </TableCell>
              </TableRow>
            </Show>
          )}
        </For>
      </Box>
    </Show>
  );
};
