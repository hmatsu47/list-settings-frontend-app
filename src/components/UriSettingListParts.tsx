import { For, Show } from "solid-js";
import Box from "@suid/material/Box";
import Button from "@suid/material/Button";
import TableCell from "@suid/material/TableCell";
import TableRow from "@suid/material/TableRow";
import { service } from "../signal";
import { formatDateTimeDisplay } from "../formatDateTime";
import { UriSetting } from "../type";
import { openUriInNewTab } from "../openUri";

const convertEnvironmentName = (environmentName: string) => {
  switch (environmentName) {
    case "stg":
      return "ステージング";
    case "prod":
      return "本番";
  }
  return environmentName;
};

type Props = {
  settings: UriSetting[] | null | undefined;
};

export const UriSettingListParts = (props: Props) => {
  const uriPrefix = localStorage.getItem("uriSettingUriPrefix");
  const uriSuffix = localStorage.getItem("uriSettingUriSuffix");

  return (
    <Show when={props.settings} fallback={<></>}>
      <For each={props.settings} fallback={<></>}>
        {(settingItem) => (
          <Show when={settingItem.service_name === service()} fallback={<></>}>
            <TableRow>
              <TableCell component="th" scope="row">
                <Button
                  variant="contained"
                  size="small"
                  color="primary"
                  onClick={() => {
                    openUriInNewTab(
                      `${uriPrefix}${settingItem.environment_name}${uriSuffix}`
                    );
                  }}
                  sx={{ textTransform: "none" }}
                  title={`${convertEnvironmentName(
                    settingItem.environment_name
                  )}環境のリリース設定画面を開く`}
                >
                  {convertEnvironmentName(settingItem.environment_name)}
                  環境設定
                </Button>
              </TableCell>
              <TableCell>
                {settingItem.next_release
                  ? `次回：${settingItem.next_release.image_uri}`
                  : settingItem.last_released
                  ? `前回：${settingItem.last_released.image_uri}`
                  : "（未設定）"}
              </TableCell>
              <TableCell>
                {settingItem.next_release
                  ? `次回：${formatDateTimeDisplay(
                      new Date(settingItem.next_release.release_at)
                    )}`
                  : settingItem.last_released
                  ? `前回：${formatDateTimeDisplay(
                      new Date(settingItem.last_released.released_at)
                    )}`
                  : "（未設定）"}
              </TableCell>
            </TableRow>
          </Show>
        )}
      </For>
    </Show>
  );
};
