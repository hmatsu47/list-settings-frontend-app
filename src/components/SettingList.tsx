import { onMount } from "solid-js";
import Box from "@suid/material/Box";
import Stack from "@suid/material/Stack";
import {
  fetchLocal,
  fetchRemote,
  fetchUriSettings,
} from "../api/fetchUriSettings";
import { fetchTagSettings } from "../api/fetchTagSettings";
import { tagSettings, uriSettings, uriRemoteSettings } from "../signal";
import { TagSettingList } from "./TagSettingList";
import { UriSettingList } from "./UriSettingList";

export const SettingList = () => {
  onMount(() => {
    if (uriSettings() === undefined) {
      fetchUriSettings(fetchLocal);
    }
    if (uriRemoteSettings() === undefined) {
      fetchUriSettings(fetchRemote);
    }
    if (tagSettings() === undefined) {
      fetchTagSettings();
    }
  });

  return (
    <Stack spacing={2} direction="column">
      <UriSettingList />
      <Box
        sx={{
          padding: "10px 10px 0 10px",
          width: "100%",
          minWidth: "1024px",
          display: "flex",
          // justifyContent: "center",
        }}
        aria-live="polite"
      />
      <TagSettingList />
    </Stack>
  );
};
