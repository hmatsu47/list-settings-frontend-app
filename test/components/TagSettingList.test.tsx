import { describe, expect, test } from "vitest";
import { render } from "@solidjs/testing-library";
import { formatSnapshot } from "../common/formatSnapshot";
import { TagSettingList } from "../../src/components/TagSettingList";
import { setTagSettings } from "../../src/signal";
import { TagSetting } from "../../src/type";

describe("<TagSettingList />", () => {
  const tagSettingList = [
    {
      title: "色設定なし・リポジトリにレコードなし",
      colors: undefined,
      header: "ヘッダータイトル1",
      uriPrefix: "http://",
      uriSuffix: "-test.example.com",
      settings: null,
      expected: ["リポジトリにイメージがありません"],
    },
    {
      title: "リポジトリにレコード1つ",
      colors: '{"dev":"#616161","prod":"#212121"}',
      header: "ヘッダータイトル2",
      uriPrefix: "http://",
      uriSuffix: "-test.example.com",
      settings: [{ environment_name: "prod", tags: ["release", "testtest"] }],
      expected: ["本番環境設定"],
    },
    {
      title: "リポジトリにレコード2つ",
      colors: '{"dev":"#616161","prod":"#212121"}',
      header: "ヘッダータイトル3",
      uriPrefix: "http://",
      uriSuffix: "-test.example.com",
      settings: [
        { environment_name: "dev", tags: ["hogefuga", "latest"] },
        { environment_name: "prod", tags: ["release", "testtest"] },
      ],
      expected: ["開発環境設定", "本番環境設定"],
    },
  ];
  beforeEach(() => {
    localStorage.removeItem("tagButtonColor");
    localStorage.removeItem("tagSettingsHeaderTitle");
    localStorage.removeItem("tagSettingUriPrefix");
    localStorage.removeItem("tagSettingUriSuffix");
  });
  afterEach(() => {
    localStorage.removeItem("selectedService");
  });
  tagSettingList.forEach((testCase) => {
    test(testCase.title, async () => {
      if (testCase.colors) {
        localStorage.setItem("tagButtonColor", testCase.colors);
      }
      localStorage.setItem("tagSettingsHeaderTitle", testCase.header);
      localStorage.setItem("tagSettingUriPrefix", testCase.uriPrefix);
      localStorage.setItem("tagSettingUriSuffix", testCase.uriSuffix);
      setTagSettings(testCase.settings);
      const { container, findByText, unmount } = render(() => (
        <TagSettingList />
      ));
      // css の名前が動的に変わるので固定値に置換
      const html = formatSnapshot(container.innerHTML);
      expect(html).toMatchSnapshot();
      // ヘッダータイトル
      const headerTitle = (await findByText(testCase.header)) as HTMLElement;
      expect(headerTitle).toHaveTextContent(testCase.header);
      // 各ボタン（クリックしない）または「リポジトリにイメージがありません」
      testCase.expected.forEach(async (environment) => {
        const text = (await findByText(environment)) as HTMLElement;
        expect(text).toHaveTextContent(environment);
      });
      unmount();
    });
  });
});
