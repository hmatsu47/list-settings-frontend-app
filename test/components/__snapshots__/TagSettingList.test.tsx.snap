// Vitest Snapshot v1

exports[`<TagSettingList /> > リポジトリにレコード1つ 1`] = `"<div aria-live=\\"polite\\" class=\\"MuiBox-root css-xxxxxxxx\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><h5 class=\\"MuiTypography-root MuiTypography-h5 MuiTypography-root css-xxxxxxxx\\">ヘッダータイトル2</h5><div class=\\"MuiTableContainer-root css-xxxxxxxx\\"><table aria-label=\\"simple table\\" class=\\"MuiTable-root css-xxxxxxxx\\"><thead class=\\"MuiTableHead-root css-xxxxxxxx\\"><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><td scope=\\"col\\" class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">設定　　　　　　　　　　</td><td scope=\\"col\\" class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">リリース対象イメージタグ　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　</td></tr></thead><tbody class=\\"MuiTableBody-root css-xxxxxxxx\\"><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th scope=\\"row\\" class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><button type=\\"button\\" class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" title=\\"本番環境のリリース設定画面を開く\\" tabindex=\\"0\\">本番環境設定<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-root css-xxxxxxxx\\">release, testtest</p></td></tr></tbody></table></div></div></div>"`;

exports[`<TagSettingList /> > リポジトリにレコード2つ 1`] = `"<div aria-live=\\"polite\\" class=\\"MuiBox-root css-xxxxxxxx\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><h5 class=\\"MuiTypography-root MuiTypography-h5 MuiTypography-root css-xxxxxxxx\\">ヘッダータイトル3</h5><div class=\\"MuiTableContainer-root css-xxxxxxxx\\"><table aria-label=\\"simple table\\" class=\\"MuiTable-root css-xxxxxxxx\\"><thead class=\\"MuiTableHead-root css-xxxxxxxx\\"><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><td scope=\\"col\\" class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">設定　　　　　　　　　　</td><td scope=\\"col\\" class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">リリース対象イメージタグ　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　</td></tr></thead><tbody class=\\"MuiTableBody-root css-xxxxxxxx\\"><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th scope=\\"row\\" class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><button type=\\"button\\" class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" title=\\"開発環境のリリース設定画面を開く\\" tabindex=\\"0\\">開発環境設定<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-root css-xxxxxxxx\\">hogefuga, latest</p></td></tr><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th scope=\\"row\\" class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><button type=\\"button\\" class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" title=\\"本番環境のリリース設定画面を開く\\" tabindex=\\"0\\">本番環境設定<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-root css-xxxxxxxx\\">release, testtest</p></td></tr></tbody></table></div></div></div>"`;

exports[`<TagSettingList /> > 色設定なし・リポジトリにレコードなし 1`] = `"<div aria-live=\\"polite\\" class=\\"MuiBox-root css-xxxxxxxx\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><h5 class=\\"MuiTypography-root MuiTypography-h5 MuiTypography-root css-xxxxxxxx\\">ヘッダータイトル1</h5><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">リポジトリにイメージがありません</h6></div></div>"`;
